### PR TITLE
Aggressively GC clusters that are in an error state.

### PR DESCRIPTION
**Pull request by jlewi**
Related to #480.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/485)

### CODE REVIEW

File: py/kubeflow/testing/cleanup_ci.py [hunk 1]
```diff
@@ -1088,6 +1103,19 @@ def main():
 
   add_deployments_args(parser_deployments)
   parser_deployments.set_defaults(func=cleanup_deployments)
+
+  ######################################################
+  # Parser for clusters
+  parser_clusters = subparsers.add_parser(
+    "clusters", help="Cleanup clusters")
+
+  parser_clusters.add_argument(
```
Comment (lluunn) [inline py/kubeflow/testing/cleanup_ci.py]: we have this already?
https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/cleanup_ci.py#L970

File: py/kubeflow/testing/cleanup_ci.py [hunk 2]
```diff
@@ -881,6 +881,8 @@ def cleanup_clusters(args):
 
   expired = []
   unexpired = []
+  stopping = []
```
Comment (gabrielwen) [inline py/kubeflow/testing/cleanup_ci.py]: could you help correct the log in LN 875?

File: py/kubeflow/testing/cleanup_ci.py [hunk 3]
```diff
@@ -904,7 +906,19 @@ def cleanup_clusters(args):
       insert_time_utc = insert_time + datetime.timedelta(hours=-1 * hours_offset)
       age = datetime.datetime.utcnow()- insert_time_utc
 
-      if age > datetime.timedelta(hours=args.max_age_hours):
+      if c.get("status", "") == "ERROR":
```
Comment (gabrielwen) [inline py/kubeflow/testing/cleanup_ci.py]: should we also GC `degraded` clusters?
https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.Status

File: py/kubeflow/testing/cleanup_ci.py (hunk 2 above)
Comment (jlewi) [reply py/kubeflow/testing/cleanup_ci.py]: Done

File: py/kubeflow/testing/cleanup_ci.py (hunk 3 above)
Comment (jlewi) [reply py/kubeflow/testing/cleanup_ci.py]: Done.

File: py/kubeflow/testing/cleanup_ci.py (hunk 1 above)
Comment (jlewi) [reply py/kubeflow/testing/cleanup_ci.py]: That's adding subparser commands and is only for the deployments command. Here we are are doing it for the clusters command.

**Pull request merged**